### PR TITLE
Issue #22 - Add pre-commit

### LINKED ISSUE #22: As a developer I want hooks to enforce code quality checks, formatter execution, and unit tests
## 💪 Motivation
...so that we can provide a guaranteed baseline of quality when a new pull request is created.

## ⚙️ Engineering Details
There's a few approaches that we could take here including just normal git hooks. I've had good luck using [pre-commit](https://pre-commit.com/) to manage this. It's pip install-able and makes the handling of these trivial. We could provide a default config file that automatically runs a formatter, linter, and executes unit tests on commit or push.

### PR DESCRIPTION
**Pull request by MJJoyce**
Add pre-commit and configuration to run mypy, black, flake8, and pytest
automatically on commit / push.

Resolve #22

### CODE REVIEW

**Comment by MJJoyce** [comment]
This is blocked by #24 and #26. This will need rebased and `pre-commit` executed on all the files to baseline our template repo prior to merging.

File: README.md [hunk 1]
```diff
@@ -33,6 +33,15 @@ To execute, run:
 
 To develop this project, use your favorite text editor, or an integrated development environment with Python support, such as [PyCharm](https://www.jetbrains.com/pycharm/).
 
+### Installation
+
+Install in editable mode and with extra developer dependencies:
+
+    pip install -e .[dev]
```
Comment (nutjob4life) [inline README.md:40]: In documentation, I prefer to see the long form command-line options (`--editable`). It feels more "documenty" that way.

**Review by nutjob4life — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (MJJoyce) [reply README.md]: done!

**Comment by MJJoyce** [comment]
Rebased to address conflicts. Change request integrated in previous commit. Merging for other work. Sean, if I missed something here please yell at me (and sorry in advance).

**Pull request merged**